src/modifierAffinityDamage.ts

```diff
@@ -0,0 +1,20 @@
+import { registerModifiers } from "./cards";
+import { getOrInitModifier } from "./cards/util";
+import * as Unit from './entity/Unit';
+import Underworld from './Underworld';
+
+export const affinityDamageId = 'Affinity: Damage';
+
+export default function registerDamageAffinity() {
+  registerModifiers(affinityDamageId, {
+    description: i18n('Damage spells cost [quantity]% less mana'),
+    costPerUpgrade: 40,
```

Comment:
IMO, Affinity runes should cost more than 40.  A cost of 40 means they could max it out at a 50% reduction after only 2 turns (200 skill points).  That seems OP.  
This is what I would recommend:
Blessing: 40
Curse: 50
Movement: 40
Soul: 70
Targeting: 80
Damage: 80

Thoughts?

Reply:
Hmm... Yeah, 40 is a little low. Here's what I'm thinking:

Generally speaking, 200 points into an affinity should be better for casting spells of that category/playstyle than 200 points into mana, and they should provide roughly equal power when part of their best synergies.

Things to consider
- Max mana
  - Helps with all categories of spells
  - Noncommittal: You can buy as much or as little as wanted, doesn't lock you into one playstyle/category of spells, easier to take early on
  - Scales with effects that rely on max mana
  - Scales well with rune effects like Mana Barrier
- Affinity 
  - Much larger effect on the specific category of spell
  - Committal: Best value at max purchase (200 points), locks you into playing around that category of spells
  - Does not scale with effects that rely on max mana
  - Scales better with flat mana recovery like from potions or mana steal
 
The balance of affinity runes vs mana is **100%** dependent on what other effects get added into the game, and every mana-related effect we add will make on or the other more valuable than before.

As far as specific values go, I like the idea of all affinities costing the same amount, and that would be balanced by what other runes are available along that playstyle. I'd also want the costs to stay semi-low. 80 feels too high because you spend half the game getting the reduced mana cost, and that doesn't leave much time for combining other "cooler" rune effects. Especially since affinities "all or nothing" compared to other runes, which can be bought once to enable some kind of combo.

I could bump them all to 60 and we balance from there? Oh, and I did leave mana affinity out. I don't really want to, because I love the mana manipulation spells and playstyle, but I'm not sure how to approach it because any changes or added rune effects seem prone to enabling infinite combos

Reply:
Hmm okay good points.  
I kind of agree, but the damage category feels different to me.  I just think getting to 50% on damage within the first two levels for instance feels super OP.  

The mana upgrade only gives you 5 mana and costs 30; so it would take 12 clicks to double from 60 to 120 (3.6 levels cleared).

Let's try having the affinities cost 60 and go from there (tho I do imagine no one's gunna pick a Blessing Affinity for that cost cause it's just not as powerful as a damage affinity

Reply:
That's true... I'm hoping all of the armor/thorns/similar effects are going to make blessings competitive. In this PR I'd actually argue that MOVEMENT is the strongest category. Liquidmancer / Heavy Impact with all the extra mana/cost reduction you can get is insane.

The soul category is an important one to keep an eye on too, with all the weird stuff it can do. And summons.